Doc/using/configure.rst

```diff
@@ -747,6 +747,17 @@ Compiler flags
    extensions.  Use it when a compiler flag should *not* be part of the
    distutils :envvar:`CFLAGS` once Python is installed (:issue:`21121`).
 
+   In particular, :envvar:`CFLAGS` should not contain:
+
+   * the compiler flag `-I` (for setting the search path for include files).
+     The `-I` flags are processed from left to right, and any flags in
+     :envvar:`CFLAGS` would take precedence over user- and package-supplied `-I`
+     flags.
```

Comment:
Likewise to https://github.com/python/cpython/pull/92754/files#r895970237, since `python-config` does honor compile-time `CPPFLAGS`, some of the search directories that we add for Python's dependencies may be required for packages as well.